tc/sgx/trusted_worker_manager/enclave/work_order_preprocessed_keys.cpp

```diff
@@ -0,0 +1,185 @@
+/* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "crypto.h"
+#include "tcf_error.h"
+#include "error.h"
+#include "jsonvalue.h"
+#include "parson.h"
+#include "json_utils.h"
+#include "enclave_utils.h"
+#include "work_order_preprocessed_keys.h"
+
+void WorkOrderPreProcessedKeys::Unpack(const JSON_Object* keys_object,
+    tcf::WorkOrderData& data_keys, std::string& encrypted_data_key) {
+
+    data_keys.index = GetJsonNumber(keys_object, "index");
+    encrypted_data_key = GetJsonStr(keys_object, "key");
+    // Decrypt base64 encoded data key using symmetric key(after decryption)
+    // from work_order_key_info json
+    data_keys.decrypted_data = DecryptKey(this->sym_key,
+        Base64EncodedStringToByteArray(encrypted_data_key));
+}  // WorkOrderPreProcessedKeys::Unpack
+
+ByteArray WorkOrderPreProcessedKeys::DecryptKey(
+    const ByteArray& encryption_key, const ByteArray& key_to_decrypt) {
```

Comment:
Either rename encryption_key as sym_key or call it decryption_key. We decrypt data with either symmetric key or decryption key

`WorkOrderPreProcessedKeys::DecryptKey(
    const ByteArray& sym_key, const ByteArray& encrypted_data_key)`

Reply:
Done